conformity/fields/currency.py

```diff
@@ -144,7 +178,152 @@ def __init__(
+    description = attr.ib(default=None, validator=attr_is_optional(attr_is_string()))  # type: Optional[six.text_type]
+
+    def __attrs_post_init__(self):  # type: () -> None
+        if not self.valid_currencies:
+            self.valid_currencies = DEFAULT_CURRENCY_CODES
+
+    def errors(self, value):  # type: (AnyType) -> ListType[Error]
+        if not isinstance(value, six.text_type):
+            return [Error('Not a unicode string currency amount')]
+
+        parts = self._format.split(value)
+        if len(parts) != 2:
+            return [Error('Currency string does not match format CUR,1234 or CUR:1234')]
+
+        currency = parts[0]
+        try:
+            value = int(parts[1])
+        except ValueError:
+            return [Error('Currency amount {} cannot be converted to an integer'.format(parts[1]))]
+
+        return _get_errors_for_currency_amount(
+            currency,
+            value,
+            self.valid_currencies,
+            self.gt,
+            self.gte,
+            self.lt,
+            self.lte,
+        )
+
+    def introspect(self):  # type: () -> Introspection
+        return strip_none({
+            'type': self.introspect_type,
+            'description': self.description,
+            'valid_currencies': (
+                '(all currencies)' if self.valid_currencies is DEFAULT_CURRENCY_CODES else sorted(self.valid_currencies)
+            ),
+            'gt': self.gt,
+            'gte': self.gte,
+            'lt': self.lt,
+            'lte': self.lte,
+        })
+
+
+class AmountResponseDictionary(Dictionary):
+    """
+    Conformity field that ensures that the value is a dictionary containing at least fields `'currency'` and `'value'`
+    and optionally fields `'major_value'` and `'display'`. This field requires that Currint be installed.
+    """
+
+    def __init__(self, description=None, major_value_required=True, display_required=True):
+        # type: (Optional[six.text_type], bool, bool) -> None
+        """
+        Construct the field.
+
+        :param description: The description for documentation
+        :param major_value_required: By default, `'major_value'` is a required field in the response, but setting this
+                                     to `False` makes it optional
+        :param display_required: By default, `'display'` is a required field in the response, but setting this to
+                                 `False` makes it optional
+        """
+        optional_keys = ()  # type: TupleType[six.text_type, ...]
+        if not major_value_required:
+            optional_keys += ('major_value', )
+        if not display_required:
+            optional_keys += ('display', )
+        super(AmountResponseDictionary, self).__init__(
+            {
+                'currency': Constant(*DEFAULT_CURRENCY_CODES),
+                'value': Integer(),
+                'major_value': UnicodeString(),
```

Comment:
Not sure if it's worth doing it, but not any major_value is valid for any currency, especially for currencies that don't have decimal (ie CLP, JPY). currint can validate the major value for that case.

Reply:
Well this field is intended only for responses (hence why it doesn’t have `valid_currencies`, `gt`, `lt`, etc., like the request field), so presumably any value put into it has come directly from a `currint.Amount` instance and, therefore, is already valid.